src/atomic64.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2014 The Prometheus Authors
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use atomic::{Atomic, Ordering};
+
+pub struct AtomicF64 {
+    inner: Atomic<f64>,
+}
+
+impl AtomicF64 {
+    pub fn new(val: f64) -> AtomicF64 {
+        AtomicF64 { inner: Atomic::new(val) }
+    }
+
+    #[inline]
+    pub fn get(&self) -> f64 {
+        self.inner.load(Ordering::Relaxed)
```

Comment:
I think it's better to use `Ordering::Acquire`, but @ngaut suggests `Ordering::Relaxed`.